js/pokemon.js

```diff
@@ -100,44 +126,36 @@ $(document).ready(function() {
  * Set the range of numbers to generate from depending on the Pokemon Generation selected
  */
  
-function setGen(selectedGen) {
-
-    if (selectedGen == 1) {
-        minPokemonNumber = 1;
-        maxPokemonNumber = 151;
-    } else if (selectedGen == 2) {
-        minPokemonNumber = 152;
-        maxPokemonNumber = 251;
-    } else if (selectedGen == 3) {
-        minPokemonNumber = 252;
-        maxPokemonNumber = 386;
-    } else if (selectedGen == 4) {
-        minPokemonNumber = 387;
-        maxPokemonNumber = 493;
-    } else if (selectedGen == 5) {
-        minPokemonNumber = 494;
-        maxPokemonNumber = 649;
-    } else {
-        minPokemonNumber = 1;
-        maxPokemonNumber = 649;
-    }
+function setGen(genToAffect) {
 
-    if (newGen == -1) {
-        // first time this has been called, so make the choice active, not just selected
-        document.getElementById('gen' + selectedGen).className += " current";
-        currentGen = selectedGen;
-    } else {
-        document.getElementById('gen' + newGen).className = document.getElementById('gen' + newGen).className.replace('selected','');
+
+    genToAffect = parseInt(genToAffect);
+
+    //Before editing gen selection, we ensure that the user is not about to remove their last gen
+    if (!(newGen.length === 1 && newGen[0] === genToAffect)) {
+
+        //Remove all the "current" classes and replace them with "selected" classes if it's the user's first time clicking a gen this round
+        if ($('.genSelect').hasClass('current')) {
```

Comment:
Functionally it doesn't make a difference, I thought it better not to looking for elements to change if they don't exist, but if it's better practice to remove all extraneous lines of code, then get rid of it.